### PR TITLE
Fix Panic generated when user does not have a home directory set

### PR DESCRIPTION
**Pull request by vibhansa-msft**
- If user does not have a home directory, blobfuse panics
- When user provides "default-working-dir" it is not honored while defining default logging path
- At init time default logging is set to syslog, still blobfuse tries to create a directory under user home directory for logs.
- Mount path or logging path with any environment variable set fails. e.g. mountpath: $HOME/abc

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -6,6 +6,8 @@
 - [csi-driver #809](https://github.com/kubernetes-sigs/blob-csi-driver/issues/809) Fail to mount when uid/gid are provided on command line.
 - [#1032](https://github.com/Azure/azure-storage-fuse/issues/1032) `mount all` CLI params parsing fix when no `config-file` and `tmp-path` is provided.
 - [#1015](https://github.com/Azure/azure-storage-fuse/issues/1015) Default value of `ignore-open-flags` config parameter changed to `true`
+- [#1036](https://github.com/Azure/azure-storage-fuse/issues/1036) Panic if user does not have a $HOME directory set.
```
Comment (tasherif-msft) [inline CHANGELOG.md:9]: The changelog entry here is kind of misleading - we should say something like "fix for panicking when $HOME dir is not set and respect `--default-working-dir` cli param setting"

File: cmd/mount.go [hunk 2]
```diff
@@ -110,9 +110,22 @@ func (opt *mountOptions) validate(skipEmptyMount bool) error {
...
 		return fmt.Errorf("invalid log level [%s]", err.Error())
 	}
-	opt.Logging.LogFilePath = os.ExpandEnv(opt.Logging.LogFilePath)
+
+	if opt.DefaultWorkingDir != "" {
+		common.DefaultWorkDir = opt.DefaultWorkingDir
+
+		if opt.Logging.LogFilePath == common.DefaultLogFilePath {
+			// If default-workng-dir is set then default log path shall be set to that path
+			// Ignore if specific log-path is provided by user
```
Comment (tasherif-msft) [inline cmd/mount.go:119]: in our readme we should mention that this param overwrites the default working dir when it comes to logging

Comment (vibhansa-msft) [reply cmd/mount.go:119]: That was the expectation earlier as well, it's just that it was not honored in GA release.

File: CHANGELOG.md (hunk 1 above)
Comment (vibhansa-msft) [reply CHANGELOG.md]: Updated

File: .gitignore [hunk 3]
```diff
@@ -20,3 +20,4 @@ test/manual_scripts/cachetest.go
 lint.log
 azure-storage-fuse
 bfusemon
+test/scripts/dirIterate.go
```
Comment (gapra-msft) [inline .gitignore:23]: what is this ignore file for?

Comment (vibhansa-msft) [reply .gitignore:23]: This was some local file getting checked-in in all my commits so just added it to the ignore list.

**Review by tasherif-msft — APPROVED** [review]

File: TSG.md [hunk 4]
```diff
@@ -105,6 +105,7 @@ If Blobfuse2 pid is listed in the output then OOM has sent a SIGKILL to Blobfuse
 
 For HNS account, always add `type: adls` under `azstorage` section in your config file. Avoid using `endpoint` unless your storage account is behind a private endpoint. Blobfuse2 uses both blob and dfs endpoints to connect to storage account. User has to expose both these endpoints over private-endpoint for blobfuse2 to function properly.
 
+To create a private-endpoint for DFS in Azure portal: Go to your storageaccount -> Networking -> Private Endpoint connections. Click + Private endpoint, fill in subscription, resourcegroup, Name, Network Interface Name and Region. Click next and under Target sub-resource select dfs. Click Virtual network and select virtual netork and Subnet. Click DNS. Select Yes for Integrate with private DNS. Select the subscription and resourcegroup for your private link DNS. Select Next, Next and select create
```
Comment (souravgupta-msft) [inline TSG.md:108]:
```suggestion
To create a private-endpoint for DFS in Azure portal: Go to your storage account -> Networking -> Private Endpoint connections. Click `+ Private endpoint`, fill in Subscription, Resource Group, Name, Network Interface Name and Region. Click next and under Target sub-resource select `dfs`. Click Virtual network and select virtual network and Subnet. Click DNS. Select Yes for Integrate with private DNS. Select the Subscription and Resource Group for your private link DNS. Select Next, Next and select Create.
```

File: cmd/mount.go [hunk 5]
```diff
@@ -110,9 +110,22 @@ func (opt *mountOptions) validate(skipEmptyMount bool) error {
 	if err := common.ELogLevel.Parse(opt.Logging.LogLevel); err != nil {
 		return fmt.Errorf("invalid log level [%s]", err.Error())
 	}
-	opt.Logging.LogFilePath = os.ExpandEnv(opt.Logging.LogFilePath)
+
+	if opt.DefaultWorkingDir != "" {
+		common.DefaultWorkDir = opt.DefaultWorkingDir
+
+		if opt.Logging.LogFilePath == common.DefaultLogFilePath {
+			// If default-workng-dir is set then default log path shall be set to that path
```
Comment (souravgupta-msft) [inline cmd/mount.go:118]:
```suggestion
			// If default-working-dir is set then default log path shall be set to that path
```

**Review by souravgupta-msft — APPROVED** [review]

**Review by gapra-msft — COMMENTED** [review]
github required text

**Pull request merged**